File: src/optic/lua/ssa/instructions/Comment.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;

public final class Comment implements Step {
	private final String text;

	public Comment(String text) {
		this.text = text;
	}

	@Override
	public String toString() {
		return "// " + text;
	}
}

File: src/optic/lua/asm/instructions/Comment.kt
package optic.lua.asm.instructions

import optic.lua.asm.Step
import optic.lua.asm.StepVisitor

class Comment(val text: String) : Step {
    override fun <T : Any, X : Throwable> accept(visitor: StepVisitor<T, X>): T = visitor.visitComment(text)

    override fun toString(): String {
        return "// $text"
    }
}
